extension-iac/src/iac_android.cpp

```diff
@@ -284,59 +198,36 @@ dmExtension::Result InitializeIAC(dmExtension::Params* params)
 
 dmExtension::Result FinalizeIAC(dmExtension::Params* params)
 {
-    if (params->m_L == g_IAC.m_Listener.m_L && g_IAC.m_Listener.m_Callback != LUA_NOREF) {
-        dmScript::Unref(g_IAC.m_Listener.m_L, LUA_REGISTRYINDEX, g_IAC.m_Listener.m_Callback);
-        dmScript::Unref(g_IAC.m_Listener.m_L, LUA_REGISTRYINDEX, g_IAC.m_Listener.m_Self);
-        g_IAC.m_Listener.m_L = 0;
-        g_IAC.m_Listener.m_Callback = LUA_NOREF;
-        g_IAC.m_Listener.m_Self = LUA_NOREF;
+    if (params->m_L == dmScript::GetCallbackLuaContext(g_IAC.m_Listener)) {
+        dmScript::DestroyCallback(g_IAC.m_Listener);
+        g_IAC.m_Listener = 0;
     }
     return dmIAC::Finalize(params);
 }
 
-
-dmExtension::Result UpdateIAC(dmExtension::Params* params)
+static void IAC_OnCommand(IACCommand* cmd, void*)
 {
-    if (g_IAC.m_CmdQueue.Empty())
+    switch (cmd->m_Command)
     {
-        return dmExtension::RESULT_OK;
-    }
+    case IAC_INVOKE:
+        HandleInvocation(cmd);
+        break;
 
-    DM_MUTEX_SCOPED_LOCK(g_IAC.m_Mutex);
+    default:
+        assert(false);
+    }
 
-    for (uint32_t i=0;i!=g_IAC.m_CmdQueue.Size();i++)
-    {
-        IACCommand& cmd = g_IAC.m_CmdQueue[i];
-
-        switch (cmd.m_Type)
-        {
-            case CMD_INVOKE:
-                {
-                    if (g_IAC.m_Listener.m_Callback == LUA_NOREF)
-                    {
-                        dmLogError("No iac listener set. Invocation discarded.");
-                        g_IAC.m_StoredInvocation.Store(cmd.m_Payload, cmd.m_Origin);
-                    }
-                    else
-                    {
-                        OnInvocation(cmd.m_Payload, cmd.m_Origin);
-                    }
-                }
-                break;
-        }
-        if (cmd.m_Payload != 0x0)
-        {
-            free((void*)cmd.m_Payload);
-            cmd.m_Payload = 0x0;
-        }
-        if (cmd.m_Origin != 0x0)
-        {
-            free((void*)cmd.m_Origin);
-            cmd.m_Origin = 0x0;
-        }
+    if (cmd->m_Payload) {
+        free(cmd->m_Payload);
```

Comment:
Fyi, no need to check the pointer if it's zero or not since we memset the entire struct to 0.

Reply:
👍